### PR TITLE
Fix transcript diagram errors

### PR DESCRIPTION
**Pull request by imransl**
## Type

- [x] Bug fix
- [ ] New feature
- [ ] Improvement to existing feature

## Related JIRA Issue(s)
[ENSWBSITES-1248](https://www.ebi.ac.uk/panda/jira/browse/ENSWBSITES-1248)

## Description
There were some issues in the transcript diagram in Entity viewer such as, gaps between some introns and exons as well as introns running into exons. I've fixed these issues to some extent, however the latter is still present in some places albeit to a lesser extent. A screenshot of a list of these transcripts is below.

Before the changes:

image

After the changes:

image

## Deployment URL
http://transcript-diagram-errors.review.ensembl.org/

## Views affected
Entity viewer -> transcript

### CODE REVIEW

File: src/ensembl/src/content/app/entity-viewer/gene-view/components/unspliced-transcript/UnsplicedTranscript.tsx [hunk 1]
```diff
@@ -132,20 +124,14 @@ const Backbone = (
...
-    }
 
-    if (i === spliced_exons.length - 1) {
-      intervals.push([exonEnd, transcriptLength]);
+    if (i < spliced_exons.length - 1) {
+      const {
+        relative_location: { start: nextExonStart }
+      } = spliced_exons[i + 1];
+      intervals.push([exonEnd, nextExonStart]);
     }
```
Comment (imransl) [inline src/ensembl/src/content/app/entity-viewer/gene-view/components/unspliced-transcript/UnsplicedTranscript.tsx:135]: I tried to simplify this a bit. I didn't see a difference between the transcript diagram with the old code and the new one, so perhaps this works. However, would like to know what you think about this.

File: src/ensembl/src/content/app/entity-viewer/gene-view/components/unspliced-transcript/UnsplicedTranscript.tsx [hunk 2]
```diff
@@ -54,11 +54,8 @@ It's possible that later we will switch to using UTRs for this purpose
 */
 
 const UnsplicedTranscript = (props: UnsplicedTranscriptProps) => {
-  const {
-    spliced_exons,
-    product_generating_contexts,
-    slice
-  } = props.transcript;
+  const { spliced_exons, product_generating_contexts, slice } =
+    props.transcript;
```
Comment (azangru) [inline src/ensembl/src/content/app/entity-viewer/gene-view/components/unspliced-transcript/UnsplicedTranscript.tsx:58]: Is this what Prettier is doing? I hate Prettier!

Comment (imransl) [reply src/ensembl/src/content/app/entity-viewer/gene-view/components/unspliced-transcript/UnsplicedTranscript.tsx:58]: Yes it is the culprit once more

**Comment by azangru** [comment]
Superseded by https://github.com/Ensembl/ensembl-client/pull/565

**Pull request closed without merging**